### PR TITLE
Fix #2375 custom mod tools permission checks missing

### LINKED ISSUE #2375: Custom mod tools group permission checks missing
A continuation of https://github.com/mybb/mybb/issues/1571
They're still missing from:
- moderation.php https://github.com/mybb/mybb/blob/fda43f3dc95421a50bf2f6eb0de0a92a10f1490b/moderation.php#L203
- search.php https://github.com/mybb/mybb/blob/9d82f2a12e0461e488a6c08be693d0ee0b558016/search.php#L684

Also, `LIKE '%,-1,%'` seems to be redundant, comparison should be enough because -1 is the only value.

### PR DESCRIPTION
**Pull request by Destroy666x**
As the title says.

https://github.com/mybb/mybb/issues/2375

### CODE REVIEW

File: search.php [hunk 1]
```diff
@@ -674,22 +674,43 @@
...
-				case "sqlite":
-					$query = $db->simple_select("modtools", "tid, name", "type='t' AND (','||forums||',' LIKE '%,-1,%' OR forums='')");
-					break;
-				default:
-					$query = $db->simple_select("modtools", "tid, name", "type='t' AND (CONCAT(',',forums,',') LIKE '%,-1,%' OR forums='')");
-			}
+			$forum_stats = $cache->read("forumsdisplay");
 
-			while($tool = $db->fetch_array($query))
+			if(is_moderator(0, 'canusecustomtools') && !empty($forum_stats[-1]['modtools']))
```
Comment (Destroy666x) [inline search.php]: https://github.com/mybb/mybb/pull/2376 should be merged before this for the `is_moderator` check to work.

**Comment by Sama34** [comment]
For some reason I'm still shown the custom moderation tool even when my group is not allowed.
https://github.com/mybb/mybb/blob/feature/moderation.php#L1043-L1051
https://github.com/mybb/mybb/blob/feature/moderation.php#L202-L210

**Comment by Destroy666x** [comment]
You're right, I missed those 2 places. I'll update it after exams

**Comment by Cu8eeeR** [comment]
@Destroy666x - any way to get this fixed into MyBB 1.8.9 release? Thank you!

**Comment by Cu8eeeR** [comment]
@PaulBender @Devilshakerz @JordanMussi and others devs - since destroy666x is busy and not active can you finish this PR? Thx

**Comment by Ben-MyBB** [comment]
I'm going to take a look at fixing the conflicts here since the majority is done.

**Pull request closed without merging**